sprout/appliances/tasks.py

```diff
@@ -152,6 +152,121 @@ def kill_unused_appliances(self):
                 kill_appliance.delay(appliance.id)
 
 
+@singleton_task()
+def delete_nonexistent_appliances(self):
+    """Goes through orphaned appliances' objects and deletes them from the database."""
+    expiration_time = (timezone.now() - timedelta(**settings.ORPHANED_APPLIANCE_GRACE_TIME))
+    for appliance in Appliance.objects.filter(ready=True).all():
+        if appliance.name in redis.renaming_appliances:
+            continue
+        if appliance.power_state == Appliance.Power.ORPHANED:
+            if appliance.power_state_changed > expiration_time:
+                # Ignore it for now
+                continue
+            self.logger.info(
+                "I will delete orphaned appliance {}/{}".format(appliance.id, appliance.name))
+            try:
+                appliance.delete()
+            except ObjectDoesNotExist as e:
+                if "AppliancePool" in str(e):
+                    # Someone managed to delete the appliance pool before
+                    appliance.appliance_pool = None
+                    appliance.save(update_fields=['appliance_pool'])
+                    appliance.delete()
+                else:
+                    raise  # No diaper pattern here!
+    # If something happened to the appliance provisioning process, just delete it to remove
+    # the garbage. It will be respinned again by shepherd.
+    # Grace time is specified in BROKEN_APPLIANCE_GRACE_TIME
+    expiration_time = (timezone.now() - timedelta(**settings.BROKEN_APPLIANCE_GRACE_TIME))
```

Comment:
Almost seems like a 2nd singleton_task, one for orphaned appliances and one for broken appliances?

Reply:
yep, there even one more task. 
We have there:
1. kill unused appliances;
2. kill nonexistent appliances; 
3. kill lost appliances.

I don't want to combine everything into one task because celery prefers to have many small tasks instead of one big.